syntax = "proto3";

package schedule;

import "google/protobuf/timestamp.proto";

message Schedule {
    uint64 id = 1;
    string medicine_name = 2;
    uint64 periodicity = 3;
    bool receipt_duration_endless = 4;
    optional google.protobuf.Timestamp receipt_duration_end = 5;
    uint64 user_id = 6;
}

message ScheduleAddSchema {
    string medicine_name = 1;
    uint64 periodicity = 2;
    bool receipt_duration_endless = 3;
    optional google.protobuf.Timestamp receipt_duration_end = 4;
    uint64 user_id = 5;
}

message GetSingleScheduleRequest {
    uint64 user_id = 1;
    uint64 schedule_id = 2;
}

message GetSingleScheduleResponse {
    Schedule schedule = 1;
}

message CreateSingleScheduleRequest {
    ScheduleAddSchema schedule = 1;
}

message CreateSingleScheduleResponse {
    uint64 id = 1;
}

message UserIdRequest {
    uint64 user_id = 1;
}

message GetSchedulesListResponse {
    repeated Schedule schedules = 1;
}

message GetNextTakingsResponse {
    repeated string medicine_name = 1;
}

service ScheduleServer {
    rpc GetSingleSchedule (GetSingleScheduleRequest) returns (GetSingleScheduleResponse) {}
    rpc CreateSingleSchedule (CreateSingleScheduleRequest) returns (CreateSingleScheduleResponse) {}
    rpc GetSchedulesList (UserIdRequest) returns (GetSchedulesListResponse) {}
    rpc GetNextTakings (UserIdRequest) returns (GetNextTakingsResponse) {}
}

